doc/hash/hash-design.md

```diff
@@ -665,6 +746,19 @@ will be extended with a new common type.
             enum INNER_L4_SRC_PORT;
         }
     }
+
+    typedef hash-algorithm {
+        description "Represents hash algorithm";
```

Comment:
For hashing-algorithm and hash-field, should we consider Identity instead of enum? Identity will be better if we want to expand the hashing-algorithm list and hash-field list.

Reply:
@sallylsy what is the benefit comparing to `enum`? IMHO, this adds too much "flexibility" and complexity